### ISSUE TITLE
Can't find where the data weighting matrix (Wd) is implemented 

### ISSUE THREAD

**Issue by leonli60639404** [gimli-org/pyGIMLi#795]
### Problem description

I am trying to modify the data weighting matrix to see how that affects the inversion output, but I could not find where the data weighting matrix (Wd) is implemented in the code? Thanks. 

### Your environment

Please provide the output of `print(pygimli.Report())` here. If that does not
work, please give provide some additional information on your:

**Operating system**: e.g. Windows, Linux or Mac?
**Python version**: e.g. 3.9, 3.10, etc.?
**pyGIMLi version**: Output of `print(pygimli.__version__)`
**Way of installation**: e.g. Conda package, manual compilation from source, etc.

### Steps to reproduce

Tell us how to reproduce this issue. Ideally, you could paste the code that produces the error:

```python
import pygimli as pg
...
```

### Expected behavior

Tell us what should happen or what you want to achieve.

### Actual behavior

Tell us what happens instead and/or provide the output of your script.

```
Paste your script output here.
```

> If possible, please add one or more labels to your issue, e.g. if you expect that your issue is rather a question than a problem with the code, please add the label "question".

**Comment by halbmy** [comment]
The data weighting matrix is created from the data errors as e.g. explained in
https://www.pygimli.org/_tutorials_auto/3_inversion/plot_5_Regularization.html
so to change the weighting you will have to change the error.

**Comment by leonli60639404** [comment]
Thank you for the response. I would like to change the actual matrix into a banded matrix to investigate the effect of correlated vs uncorrelated data noise on the inverted model. Is there a way to physically implement the actual Wd matrix? Thanks!

**Comment by halbmy** [comment]
A very good question, indeed! I am afraid, this is currently not supported, but I'll put it in the list for the redesign of the inversion that the data covariance can also be specified as alternative to the error vector.

**Comment by leonli60639404** [comment]
Thank you. But for simplicity, could I simply read in a different matrix to overwrite the default Wd matrix ... ? Just a thought ...

**Comment by halbmy** [comment]
No, in the inverse solver it is currently implemented as a simple vector. But I will make it flexible.

**Comment by halbmy** [comment]
Closing as the original title is not descriptive anymore. Furthermore, we don't have non diagonal data covariance on the list right now.
